File: digital_card/lib/services/tables_service.dart
import 'dart:convert';

import 'package:digital_card/models/table_model.dart';
import 'package:digital_card/models/util_model.dart';
import "package:http/http.dart" as http;

class TablesService extends UtilModel {
  Future<List<TableModel>> getTables() async {
    try {
      http.Response response = await http.get(uri("tables"));

      if (response.statusCode == 200) {
        List<dynamic> jsonData = json.decode(response.body);
        return jsonData.map((json) => TableModel.fromJson(json)).toList();
      } else {
        handleError("No se pudo obtener las mesas: ", response.statusCode);
      }
    } catch (err) {
      handleError("Error al obtener las mesas: ", err);
    }
    return [];
  }

  Future<TableModel?> getTable(int id) async {
    try {
      http.Response response = await http.get(uri("tables/$id"));
      if (response.statusCode == 200) {
        Map<String, dynamic> jsonData = json.decode(response.body);
        return TableModel.fromJson(jsonData);
      } else {
        handleError("No se pudo obtener la mesa: ", response.statusCode);
      }
    } catch (err) {
      handleError("Error al obtener una mesa: ", err);
    }
    return null;
  }
}

File: digital_card_web/src/services/tables.service.js
class TableService {
    constructor() {
        this.url = "http://localhost:3000/api/tables";
    }

    async getTables() {
        let tables = []
        try {
            const response = await axios(this.url);
            tables = response.data.map((table) => Table.fromJson(table))
        } catch (err) {
            console.log(`Ocurrio un error: ${err}`)
        }
        return tables;
    }
}                                                                                           